File: android/app/src/main/java/com/baatcheet/app/domain/usecase/auth/GetCurrentUserUseCase.kt
package com.baatcheet.app.domain.usecase.auth

import com.baatcheet.app.domain.model.User
import com.baatcheet.app.domain.repository.AuthRepository
import javax.inject.Inject

/**
 * Get Current User Use Case
 * 
 * Retrieves the currently authenticated user.
 */
class GetCurrentUserUseCase @Inject constructor(
    private val authRepository: AuthRepository
) {
    /**
     * Get current user if authenticated
     */
    operator fun invoke(): User? {
        return authRepository.getCurrentUser()
    }
}

File: ios/iOSBaseProject/Domain/UseCases/Auth/GetCurrentUserUseCase.swift
//
//  GetCurrentUserUseCase.swift
//  BaatCheet
//
//  Domain layer - Get Current User Use Case
//

import Foundation

/// Use case for getting the currently authenticated user.
protocol GetCurrentUserUseCaseProtocol {
    func execute() -> User?
}

final class GetCurrentUserUseCase: GetCurrentUserUseCaseProtocol {
    private let repository: AuthRepositoryProtocol
    
    init(repository: AuthRepositoryProtocol) {
        self.repository = repository
    }
    
    func execute() -> User? {
        return repository.getCurrentUser()
    }
}
